get configured CRAN mirror

cderv opened the pull request:
This PR just allows to check if a CRAN mirror is set on the session before using default CRAN mirror of pkgsnap otherwise.

I hesitate to put the `cran_mirror` args in restore to allow user to set it manually. However, as he can change is _repos_ options manually. 

So I think it is fine. 

As said in the issue post, tests are not working but not due these changes I think.

cderv committed:
get configured CRAN mirror

gaborcsardi commented on R/urls.R:
```diff
@@ -1,12 +1,26 @@
+#' If a CRAN mirror is configured, use it. Otherwise, use default
+#' @keywords internal
+
+get_CRAN_mirror <- function() {
```
can you make cran lowercase please?

gaborcsardi commented on R/urls.R:
```diff
@@ -1,12 +1,26 @@
+get_CRAN_mirror <- function() {
+  repos <- getOption("repos")
+  if (!("CRAN" %in% names(repos)) || "@CRAN@" %in% repos) {
+    message("No CRAN repository configured. Using default for pkgsnap: ", default_cran_mirror)
```
I would just print `"Using CRAN mirror %s."`, always.

gaborcsardi commented:
Thanks!

cderv committed:
cran lowercase

cderv committed:
always print message for used cran mirror

cderv commented:
Thank you for the review. I modified based on your comments.
For printing message, I changed the function to **always** print a message with the selected cran mirror.
Hope it is what you meant.

cderv commented:
Is there still a problem with this PR ? 
Not sure to understand travis check issue.

Also, is pkgsnap planned to be send to CRAN ? or will it be included in other project ? 

Thanks.

gaborcsardi commented:
No, it's good. Thanks for the nudge!
